packages/flutter_tools/lib/src/android/application_package.dart

```diff
@@ -381,10 +475,19 @@ class ApkManifestData {
       return null;
     }
 
-    final _Attribute? nameAttribute = launchActivity.firstAttribute('android:name');
+    final _Attribute? nameAttribute = isActivityAlias ? launchActivity.firstAttribute('android:targetActivity') : launchActivity.firstAttribute('android:name');
+
     // "io.flutter.examples.hello_world.MainActivity" (Raw: "io.flutter.examples.hello_world.MainActivity")
     final String? activityName = nameAttribute?.value?.substring(1, nameAttribute.value?.indexOf('" '));
 
+    // Checking if Target Activity is declared
+    if (isActivityAlias && activityName != null) {
+      if (!activityNames.contains(activityName)) {
+        logger.printError('Error running $packageName. Target activity not found');
+        return null;
+      }
+    }
```

Comment:
Is this check necessary? If so, why do we do this only when parsing from an `aapt` XML dump (and not when parsing from regular XML)?

Reply:
Yes. `activity-alias` is an alias of an `activity`. So when you want to use an `activity-alias` as  `android.intent.category.LAUNCHER` and `android.intent.action.MAIN`, you must set the `targetActivity`. 

So, we need to check if it is declared or not. 


I have updated the code to check in the regular XML parsing